### PR TITLE
e2e: Move EnsureTillerInstalled and minor tidying up

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#4606

Some small fixes to break down https://github.com/giantswarm/aws-operator/pull/1992.

### CODE REVIEW

**Comment by sslavic** [comment]
FYI flattening commits removed running of e2e tests.

**Comment by rossf7** [comment]
> FYI flattening commits removed running of e2e tests.

@sslavic Ack but the PR is for `legacy` branch that still has e2e AIUI.

So can I just approve the `build/hold` step like before?

File: integration/test/draining/draining_test.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ import (
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/micrologger"
 	"github.com/spf13/afero"
-	"k8s.io/api/core/v1"
+	corev1 "k8s.io/api/core/v1"
```
Comment (rossf7) [inline integration/test/draining/draining_test.go]: Just fixing the alias.

**Comment by rossf7** [comment]
RFR. Also @TheoBrigitte @kopiczko can you confirm its safe to run the build/hold step?

**Comment by kopiczko** [comment]
> RFR. Also @TheoBrigitte @kopiczko can you confirm its safe to run the build/hold step?

It's safe but half of tests will fail (there is no current version; only wip).

**Comment by xh3b4sd** [comment]
Why do we want to do more work in `legacy` than absolutely necessary? Shouldn't we try to only use kind for e2e tests in `master`?

**Comment by rossf7** [comment]
> Why do we want to do more work in legacy than absolutely necessary? Shouldn't we try to only use kind for e2e tests in master?

Not sure as legacy is going to live a long time. But process wise I think fixes should go in master first. I'll create a fresh PR and do that.

File: integration/test/draining/draining_test.go [hunk 2]
```diff
@@ -18,12 +18,12 @@ import (
...
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/rest"
 
 	"github.com/giantswarm/aws-operator/integration/env"
-	"github.com/giantswarm/aws-operator/service/controller/legacy/v29/key"
+	"github.com/giantswarm/aws-operator/service/controller/legacy/v31/key"
```
Comment (rossf7) [inline integration/test/draining/draining_test.go]: This will sorted once the operator is flat.

**Pull request merged**